xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js

```diff
@@ -58,8 +59,19 @@ define('modal', ['jquery', 'l10n!modal', 'bootstrap'], function($, translations)
     });
     modal.find('.modal-title').text(definition.title);
     modal.find('.modal-body').html('').append(definition.content);
-    modal.find('.modal-footer .btn-primary').text(definition.acceptLabel);
+    const submitButton = modal.find('.modal-footer .btn-primary');
+    submitButton.text(definition.acceptLabel);
     modal.find('.modal-footer .btn[data-dismiss="modal"]').text(definition.dismissLabel);
+    if (definition.form) {
+      // We want the submit button to actually submit the form
+      submitButton.removeAttr('type');
+      // This modal should act and look like a form. We replace its content node by a form, so we can benefit from
+      // native form utilities, such as implicit form validation (pressing Enter on a text field)
+      const modalContent= modal.find('.modal-content');
+      const formModalContent = $('<form class="modal-content">');
+      formModalContent.append(modalContent.contents());
+      modalContent.replaceWith(formModalContent);
```

Comment:
Are we sure that there is no code targeting `div.modal-content`? I'm just asking here, it's probably fine, this is just the main place here where I could see problems with backwards compatibility.

Reply:
From what I can see on https://github.com/search?q=org%3Axwiki%20.modal-content&type=code this is okay :+1: 

It could be interesting to find out when / why we target `div` . `div` is not semantic, so IMO it would be good for it to be replaceable without worrying about breaking anything but also pointless to put it in a CSS selector or an xpath (if that's needed, it means that there's not enough semantics in the first place :) ).

Unfortunately there's [plenty of examples](https://github.com/search?q=repo%3Axwiki%2Fxwiki-platform%20%22div.%22&type=code) of this in the codebase.